index.js

```diff
@@ -10,7 +8,13 @@ var parse = require('url-parse');
  * @api public
  */
 function origin(url) {
-  if ('string' === typeof url) url = parse(url);
+  if ('string' === typeof url) {
+    try {
+      url = new URL(url);
+    } catch (er) {
+      return 'null';
+    }
+  }
 
```

Comment:
Remove everything under this line.

Reply:
Suddenly, the URL API returns null for file protocol, which breaks current tests. I left protocol logic checking to be consistent with old tests.

Reply:
`'null'` origin is expected for the `'file:'` protocol.

Reply:
[This test](https://github.com/AlexHladin/original/blob/c159ed70e9d0d0f572f147ed7fc607efb3a71675/test.js#L67) expects origin for `file:` protocolor